vertx-web-client/src/main/java/io/vertx/ext/web/client/CachingWebClientOptions.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2021 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.web.client;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author <a href="mailto:craigday3@gmail.com">Craig Day</a>
+ */
+public class CachingWebClientOptions extends WebClientOptions {
```

Comment:
properties should use get/set pattern, in addition CachingWebClientOptions should be annotated with `@DataObject`

Reply:
For boolean properties, I followed the pattern from `WebClientOptions` where instead of `getBooleanProperty` it was `isBooleanProperty`. Is it OK to leave the `is*` pattern here, or should I do `getBooleanProperty` for this class?

Reply:
I think *is* is fine but we can check the generated converter